osquery/tables/system/darwin/extended_attributes.cpp

```diff
@@ -0,0 +1,223 @@
+  do {
+    attributes.push_back(std::string(content));
+    content += attributes.back().size() + 1;
+  } while (content - stable < value);
+  free(stable);
+  return attributes;
+}
+
+// Parse a given XAttrAttribute struct as a where from plist
+void parseWhereFrom(QueryData &results,
+                    const XAttrAttribute &x_att,
+                    const std::string &path,
+                    const std::string &directory) {
+  pt::ptree data;
+  osquery::parsePlistContent(x_att.attribute_data, data);
+
+  if (data.count("root") > 0) {
+    std::vector<std::string> values;
+    for (const auto &node : data.get_child("root")) {
+      auto value = node.second.get<std::string>("", "");
+      values.push_back(value);
+    }
+    if (values.size() == 2) {
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", "where_from_download_url"},
+                         {"value", values[0]}});
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", "where_from_download_page"},
+                         {"value", values[1]}});
+    } else {
+      // Changed to blank because it might not be no data, it might be we just
+      // don't know how to parse because it's corrupted. So I have elected
+      // to not make the No Data decision for the investigator.
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", "download_url"},
+                         {"value", ""}});
+      results.push_back({{"path", path},
+                         {"directory", directory},
+                         {"key", "download_page"},
+                         {"value", ""}});
+    }
+  }
+}
+
+// Parse a given XAttrAttribute struct as a Apple Quarantine string
+void parseQuarantineFile(QueryData &results,
+                         const XAttrAttribute &x_att,
+                         const std::string &path,
+                         const std::string &directory) {
+  std::vector<std::string> values = osquery::split(x_att.attribute_data, ";");
+  if (values.size() < 2) {
+    return;
+  }
+  results.push_back({{"path", path},
+                     {"directory", directory},
+                     {"key", "quarantine_creator"},
+                     {"value", values[2]}});
+  results.push_back({{"path", path},
+                     {"directory", directory},
+                     {"key", "quarantine_state"},
+                     {"value", values[0]}});
+}
+
+// Process a file and extract all attribute information, parsed or not.
+void getFileData(QueryData &results,
+                 const std::string &path,
+                 const std::string &directory) {
+  std::vector<std::string> attributes = parseExtendedAttributeList(path);
+  for (const auto &attribute : attributes) {
+    struct XAttrAttribute x_att = getAttribute(path, attribute);
+    try {
```

Comment:
Instead of expecting to except you can use `*xParseMap.count(attribute)` for an existence check.